compute_tools/src/compute.rs

```diff
@@ -1385,3 +1391,87 @@ pub fn forward_termination_signal() {
         kill(pg_pid, Signal::SIGQUIT).ok();
     }
 }
+
+// Start a thread that will periodically renew LSN lease for static compute
+fn start_lsn_lease_for_static(compute_state: &ComputeState) {
+    let spec = compute_state.pspec.as_ref().expect("spec must be set");
+
+    let lsn = match spec.spec.mode {
+        ComputeMode::Static(lsn) => lsn,
+        _ => return,
+    };
+
+    let conn_strings = spec.pageserver_connstr.split(',');
```

Comment:
The `pageserver_connstr` changes during the lifetime of a compute, so we probably need to load this repeatedly during the lease loop rather than one time at startup.

Reply:
Yes, I made a wrong assumption... But does it also mean, that type of compute could change (Replica -> Static)? If we need to handle this too, options that I see:

1. always start lsn_lease_loop thread even for primary / replica. if it's not Static set timeout (10 secs?) to check again 
2. hook into config-changes methods (apply_config / reconfigure) and start / stop loop from there

there is also 1.b: instead of starting thread always start global tokio runtime and use it in lsn_lease & vm_monitor & more-in-future

Reply:
>But does it also mean, that type of compute could change (Replica -> Static)?

I don't think so... even if the postgres config system allows this, I hope the control plane never does this.  maybe it's worth putting some assertion in compute_ctl or postgres though, to enforce this expectation that once we start an endpoint with a fixed LSN, it should remain that way for its whole lifetime.

Reply:
06f953ad21bd343e6c9b9c4c295cd3587af3f8f7